src/bika/lims/content/abstractanalysis.py

```diff
@@ -473,8 +473,11 @@ def setResult(self, value):
         # Ensure result integrity regards to None, empty and 0 values
         val = str("" if not value and value != 0 else value).strip()
 
+        # Check if an string result is expected
+        string_result = self.getStringResult()
```

Comment:
Can we just bail out if `string_result` is True?

This would also fix https://github.com/senaite/senaite.core/issues/1614

Reply:
#1614 also mentions wrong decimal formatting of string results, that are accepted at submission firstly but are applied after verification later.
So I think we also need to bail out string results in `getFormattedResult` like
```    ...
    # 0: The result is a StringResult, return without any formatting
    string_result = self.getStringResult() is True
    if string_result:
        # return without further result handling
        return result
```
(as you suggested yesterday, @ramonski )

Reply:
> Can we just bail out if `string_result` is True?
> 
> This would also fix #1614

As @grulisco mentions, the fix for #1614 involves the function `getFormattedResult`. The change in this PR does not have any effect to that issue, except for the capture of ">" and "<" symbols. To "bail out" if `string_result` is True, I'd need to do something like follows because the result capture date is automatically assigned as well:

```python
       # Check if an string result is expected
        if self.getStringResult():
            self.getField("Result").set(self, val)
            capture_date = None
            if val and prev_result != val:
                capture_date = DateTime()
            self.setResultCaptureDate(capture_date)
            return
        
        # UDL/LDL directly entered in the results field
        if val[:1] in [LDL, UDL]:

```


and imho this is uglier.